core/integration/features/liquidity-provision/0044-LIME-028.feature

```diff
@@ -0,0 +1,122 @@
+      | 0.01        | 0.5                          | 10                                  | 1                             | 1.0                    |
+    And the markets:
+      | id        | quote name | asset | risk model          | margin calculator         | auction duration | fees          | price monitoring   | data source config     | linear slippage factor | quadratic slippage factor | sla params |
+      | ETH/DEC21 | ETH        | ETH   | simple-risk-model-1 | default-margin-calculator | 1                | fees-config-1 | price-monitoring-1 | default-eth-for-future | 0.5                    | 0                         | SLA        |
+    And the parties deposit on asset's general account the following amount:
+      | party  | asset | amount  |
+      | party1 | ETH   | 100000000  |
+      | party3 | ETH   | 100000000  |
+      | party4 | ETH   | 100000000  |
+    And the parties submit the following liquidity provision:
+      | id  | party  | market id | commitment amount | fee   | lp type     |
+      | lp1 | party1 | ETH/DEC21 | 1000              | 0.001 | submission  |
+    And the parties place the following orders:
+      | party  | market id | side | volume | price | resulting trades | type       | tif     | reference     |
+      | party3 | ETH/DEC21 | buy  | 1000   | 900   | 0                | TYPE_LIMIT | TIF_GTC | p3b1          |
+      | party3 | ETH/DEC21 | buy  | 10     | 1000  | 0                | TYPE_LIMIT | TIF_GTC | p3b2          |
+      | party4 | ETH/DEC21 | sell | 1000   | 1100  | 0                | TYPE_LIMIT | TIF_GTC | p4s1          |
+      | party4 | ETH/DEC21 | sell | 10     | 1000  | 0                | TYPE_LIMIT | TIF_GTC | p4s2          |
+
+
+
+  Scenario: 003, LP is covered fully with ICEBERG orders
+    When the opening auction period ends for market "ETH/DEC21"
+    And the auction ends with a traded volume of "10" at a price of "1000"
+    Then the market data for the market "ETH/DEC21" should be:
+      | mark price | trading mode            | target stake | supplied stake | open interest | best static bid price | static mid price | best static offer price |
+      | 1000       | TRADING_MODE_CONTINUOUS | 1000         | 1000           | 10            | 900                   | 1000             | 1100                    |
+
+    # Place ICEBERG orders to cover our commitment
+    When the parties place the following iceberg orders:
+      | party  | market id | side | volume | price | resulting trades | type       | tif     | peak size | minimum visible size | only |
+      | party1 | ETH/DEC21 | buy  | 20000  | 999   | 0                | TYPE_LIMIT | TIF_GTC | 100       | 10                   | post |      
+      | party1 | ETH/DEC21 | sell | 20000  | 1001  | 0                | TYPE_LIMIT | TIF_GTC | 100       | 10                   | post |      
+
+    # Move forward an epoch and make sure the accounts do not change as we have the full epoch covered with ICEBERG orders
```

Comment:
If the iceberg peak of 100 covers the commitment, so does the full iceberg volume so if by accident the full iceberg volume is being taken into account the test will still pass. So maybe we need the test to be such that their commitment *isn't* covered by the iceberg peaks but is by the full volume, and then check they are penalised.

Reply:
Added test to cover this case and it found a bug. I've made the change to fix it in this PR. Thanks Will.